### PR TITLE
Compiler warnings

### PR DESCRIPTION
**Pull request by namtsui**
I came across some compiler warnings while getting ready to submit a port for ktx for OpenBSD. @tdm4 is the maintainer of mvdsv on OpenBSD and let me know about the PR.

This fixes:
-  Credit goes to @tdm4 for fixing the abs() issue here: https://github.com/deurk/mvdsv/commit/5ff1935dca0a839d13cb0de12864b26117b97d1b
- warning: implicit conversion from 'int' to 'float' changes value from 99999999 to 1.0E+8 [-Wimplicit-int-float-conversion]
- warning: too many arguments in call to 'FireAmbient'

warnings here: https://gist.github.com/namtsui/e5edb677c0a70bc514df1614affb971c

These are not fixed but seem harmless: warning: illegal character encoding in string literal

### CODE REVIEW

**Comment by namtsui** [comment]
OpenBSD clang version 10.0.1

File: src/bot_aim.c [hunk 1]
```diff
@@ -297,8 +297,8 @@ static void BotsModifyAimAtPlayerLogic(gedict_t* self)
 
 		if (g_random() < 0.8) {
 			// Randomise the amount but not the side that we're aiming on
-			yaw_rnd = (self->fb.last_rndaim[YAW] > 0 ? 1 : -1) * abs(yaw_rnd);
-			pitch_rnd = (self->fb.last_rndaim[PITCH] > 0 ? 1 : -1) * abs(pitch_rnd);
+			yaw_rnd = (self->fb.last_rndaim[YAW] > 0 ? 1 : -1) * fabs(yaw_rnd);
+			pitch_rnd = (self->fb.last_rndaim[PITCH] > 0 ? 1 : -1) * fabs(pitch_rnd);
```
Comment (tdm4) [inline src/bot_aim.c:301]: Shouldn't these be `fabsf()` since `yaw_rnd` and `pitch_rnd` are both `float` type? (`fabs()` is for `double` variables)

**Review by tdm4 — CHANGES_REQUESTED** [review]

File: src/bot_aim.c (hunk 1 above)
Comment (namtsui) [reply src/bot_aim.c:301]: good catch. while looking at the function I noticed the use of fabs in the same function. i'll change them all to fabsf.

**Comment by namtsui** [comment]
I replaced fabs with fabsf throughout the whole project. I realized that doubles are never used but floats are.

bg_lib.h notes this so bg_lib.h and bg_lib.c doesn't affect native builds. I'm not sure what virtual machines are:
> // bg_lib.h -- standard C library replacement routines used by code
> // compiled for the virtual machine
> 
> // This file is NOT included on native builds

I didn't touch things that used assembly calls to fabs:
> tools/lcc/alpha/osf/tst/paranoia.sbk:300:jsr $26,fabs
> tools/lcc/mips/irix/tst/paranoia.sbk:293:jal fabs
> tools/lcc/sparc/solaris/tst/paranoia.sbk:330:call fabs; nop


other math functions would have to replaced like ceil to ceilf. 
include/bg_lib.h:
```
// Math functions
double  ceil( double x );
double  floor( double x );
double  sqrt( double x );
double  sin( double x );
double  cos( double x );
double  atan2( double y, double x );
double  tan( double x );
int             abs( int n );
double  fabs( double x );
float   fabsf( float x );
double  acos( double x );
```

In commands.c, I saw that atof (doubles) could be replaced by strtof (floats) but did not do so. I had this patch at one point:
```
Index: src/commands.c
--- src/commands.c.orig
+++ src/commands.c
@@ -4589,7 +4589,7 @@ void ktpro_autotrack_predict_powerup( void )
 		return; // we use this function for quad and pent only, ring and suit is not interesting for us
 
 	best = NULL;
-	best_len = 99999999;
+	best_len = 99999999.0;
 
     for( p = world; (p = find_plr( p )); )
 	{
@@ -6033,7 +6033,7 @@ void callalias ()
 	}
 
 	trap_CmdArgv( 2, arg_x, sizeof( arg_x ) );
-	tm = fabs( atof(arg_x) );
+	tm = fabsf( strtof( arg_x, (char **)NULL ) );
 
 	if ( tm <= 0 || tm > ca_limit2 ) {
 		G_sprint(self, 2, "calling time can't be longer than %d seconds\n", ca_limit2);
```

Questions:
- Do you want to replace math functions (cos --> cosf)?
- Do you want to replace atof --> strtof?

**Comment by angeld29** [comment]
qvm dont have stdlib, for these functions used server api,  if u change name, will be q3asm error , symbol not found

**Comment by qqshka** [comment]
@tcsabina I think most if not all warnings fixed, this PR probably should be closed.

**Comment by namtsui** [comment]
@qqshka Thanks for looking into this and fixing.

**Pull request closed without merging**